USWDS-Site: Add instructions for adding self-hosted font

amyleadem opened the pull request:
## Preview
[Font family token page](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-custom-font-guidance/design-tokens/typesetting/font-family/#adding-a-font-to-uswds)

## Related 
Related to issue https://github.com/uswds/uswds-site/issues/1783
Request to be merged into https://github.com/uswds/uswds-site/pull/1905

## Tasks completed
- Reformatted custom fonts section to include both scenarios for adding a custom font to USWDS
    - Note: The two scenarios share some steps, so I created an include that can be pulled into each section. 
- Added a changelog entry (The date will need to be updated before merge)

mejiaj reviewed (changes requested):
Looks good! I tested:

- [x] Followed guidance to set Lato both cloud and locally
- [x] Internal links to new sections work properly
- [ ] Code style and consistency (minor change on quotes)
- [x] Ran AXE on page and confirmed no a11y issues in new sections

We might also double-check use of plain language. No major concerns, but good to be sure.

amyleadem commented on _includes/tokens/create-font-token.html:
````diff
@@ -0,0 +1,24 @@
+          // or whatever stack you want
+      ),
+    ),
+    ```
````
Updated in [beb1e97](https://github.com/uswds/uswds-site/pull/1972/commits/beb1e97b82a04c51147a7fad7d59b75bcb2b494e)

bonnieAcameron commented:
It looks great! I added a couple small changes/suggestions. I also want to point out that there is no space between the table and the "note" in the "Role Based Tokens" section.

<img width="1440" alt="Untitled" src="https://user-images.githubusercontent.com/96838068/213225392-44ef27a1-eadb-43c1-b0a6-2b6f7b234b28.png">

bonnieAcameron reviewed (changes requested):
I flagged a couple of small things, but other than that it looks good to go!

amyleadem commented on _includes/tokens/create-font-token.html:
````diff
@@ -0,0 +1,23 @@
+    ),
+    ```
+
+1. Then tell `$theme-font-type-[font type]` to use the new typeface token in your theme.  In this example, we are associating `Lato` with the `sans` font type.
````
Resolved in [afbe7da](https://github.com/uswds/uswds-site/pull/1972/commits/afbe7dad520201156a8c4c128b6f2b68f8666f92)

amyleadem commented on pages/design-tokens/typesetting/font-family.md:
```diff
@@ -243,32 +243,58 @@ Your context and coding style determine how you access USWDS family tokens in co
+
+1. Copy font files into your fonts directory
+1. Configure `$theme-font-[font type]-custom-src` to:
+    1. Tell the system where to find your font files
```
Updated in [afbe7da](https://github.com/uswds/uswds-site/pull/1972/commits/afbe7dad520201156a8c4c128b6f2b68f8666f92)

Note: I did not make this a global style because it differs from the USWDS list styles and I did not want it to affect the list component preview. Open to a global solution if a better solution exists.

amyleadem commented:
@bonnieAcameron I have made the requested updates.

 I also made some adjustments to the site note about additional info in the docs in [c78d12e](https://github.com/uswds/uswds-site/pull/1972/commits/c78d12edb78ee6bc8f8058a6447644876929ea5f):
-  moved the note down to the bottom of the page. It felt like it fit better there. Happy to switch it back if you think that's best.  
- I also removed some items from the note because we now cover it in our self-hosted font docs.

bonnieAcameron reviewed (approved):
Approved!
